Buildings/Controls/OBC/CDL/Continuous/PIDWithAutoTuning.mo

```diff
@@ -0,0 +1,515 @@
+  parameter Boolean reverseActing=true
+    "Set to true for reverse acting, or false for direct acting control action";
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_s
+    "Connector of setpoint input signal"
+    annotation (Placement(transformation(extent={{-260,-20},{-220,20}}),iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_m
+    "Connector of measurement input signal"
+    annotation (Placement(transformation(origin={0,-220},extent={{20,-20},{-20,20}},rotation=270),iconTransformation(extent={{20,-20},{-20,20}},rotation=270,origin={0,-120})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput y
+    "Connector of actuator output signal"
+    annotation (Placement(transformation(extent={{220,-20},{260,20}}),iconTransformation(extent={{100,-20},{140,20}})));
+  Buildings.Controls.OBC.CDL.Continuous.Subtract controlError "Control error (set point - measurement)"
+    annotation (Placement(transformation(extent={{-200,-10},{-180,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.IntegratorWithReset I(final k=1,
+    final y_start=xi_start) if with_I
+    "Integral term"
+    annotation (Placement(transformation(extent={{-50,-10},{-30,10}})));
+  Derivative D(
+    final k=1,
+    final T=Td_start/Nd,
+    final y_start=yd_start) if with_D
+    "Derivative term"
+    annotation (Placement(transformation(extent={{-50,60},{-30,80}})));
+  Buildings.Controls.OBC.CDL.Continuous.Subtract errP
+    "P error"
+    annotation (Placement(transformation(extent={{-140,130},{-120,150}})));
+  Buildings.Controls.OBC.CDL.Continuous.Subtract errD if with_D
+    "D error"
+    annotation (Placement(transformation(extent={{-140,60},{-120,80}})));
+  Buildings.Controls.OBC.CDL.Continuous.Subtract errI1 if with_I
+    "I error (before anti-windup compensation)"
+    annotation (Placement(transformation(extent={{-140,-10},{-120,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.Subtract errI2 if with_I
+    "I error (after anti-windup compensation)"
+    annotation (Placement(transformation(extent={{-90,-10},{-70,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.Limiter lim(
+    final uMax=yMax,
+    final uMin=yMin)
+    "Limiter"
+    annotation (Placement(transformation(extent={{120,80},{140,100}})));
+
+  Modelica.Blocks.Sources.RealExpression P(y=errP.y*k) annotation (Placement(transformation(extent={{-50,130},{-30,150}})));
+  Modelica.Blocks.Sources.RealExpression gainD(y=Td*k*errD.y) if
+                                                               with_D annotation (Placement(transformation(extent={{-106,60},{-86,80}})));
+  Modelica.Blocks.Sources.RealExpression gainI(y=errI2.y*k/Ti) if
+                                                               with_I
+                                                          annotation (Placement(transformation(extent={{-106,22},{-86,42}})));
+  Modelica.Blocks.Sources.RealExpression antWinGai(y=antWinErr.y/(k*Ni)) if
```

Comment:
antWinErr is conditionally removable and therefore the Modelica Language Standard does not allow using it in an equation. This causes the model check to fail.

Reply:
This block has been removed and we eliminate the usage of the RealExpression block